core/src/main/scala/kafka/coordinator/transaction/ProducerIdManager.scala

```diff
@@ -167,7 +167,7 @@ class RPCProducerIdManager(brokerId: Int,
       if (nextProducerId > currentProducerIdBlock.lastProducerId) {
         val block = nextProducerIdBlock.poll(maxWaitMs, TimeUnit.MILLISECONDS)
         if (block == null) {
-          throw Errors.REQUEST_TIMED_OUT.exception("Timed out waiting for next producer ID block")
```

Comment:
Where does this exception get caught? I wonder if it reaches the "unexpected error" handling in `KafkaApis`.

Reply:
I'd assume its caught in the same place as the other error since they both end up getting thrown here.

I wouldn't expect to see the request timed out get hit if the errors are caught.

Reply:
What I'm questioning is whether we want the exception to propagate all the way back to `KafakApis`, or if we want to handle it at a lower error. In other words, do we consider this an "unexpected" case?

Reply:
This poll specifically seems pretty unexpected to me since we typically either timeout or get the pid block. To hit this specific case, I think the poll itself has to fail which to me is unexpected. Whether the onTimeout error is expected is a different question though.

I'm not sure I understand what handling it lower would look like. Are you thinking just building in the response and not going to KafkaApis? (Also I still need to confirm it actually goes to kafka apis)

Reply:
Ok confirmed, it does go back up to KafkaApis. We can add some try catch blocks to build the error into the response, but what are we hoping to achieve? Are we trying to distinguish client vs server side timeouts?

Reply:
I was just trying to figure out if we wanted the "unexpected error" message to be logged or not. Usually we try to catch errors in the handler.

Reply:
Sure we can do that I suppose.